plugin-API.md

```diff
@@ -0,0 +1,209 @@
+# Description of the netavark plugin API
+
+A netavark plugin is a external binary which must implement a specific set of subcommands that will be called by podman and netavark.
+ - `create`: creates a network config
+ - `setup`: setup the network configuration
+ - `teardown`: tear down the network configuration
+ - `info`: show info about this plugin
+
+## Create subcommand
+
+The create subcommand creates a new network config for podman.
+The subcommand will receive the JSON network config via STDIN. Podman will populate
+the network name and ID before calling the plugin. The name and ID cannot be changed
+by the plugin. The driver name must also not be changed. All other config
+fields can be changed in the plugin.
+
+Other fields such as subnet and options will also be populated by podman when
+these options are set on the podman network create command, i.e. `--subnet`
+and `--option`. The plugin should validate the given values and error out for
```

Comment:
if you end up having to address any other comments, consider dropping the use of "should"; i.e. "The plugin validates (or will validate) ... "

Reply:
fixed